totalRP3/modules/NamePlates/NamePlates.lua

```diff
@@ -0,0 +1,1124 @@
+				configKey = CONFIG_NAMEPLATES_ACTIVE_QUERY,
+				dependentOnOptions = {
+					CONFIG_NAMEPLATES_ENABLE_CUSTOMIZATIONS,
+				},
+			},
+			{
+				inherit = "TRP3_ConfigCheck",
+				title = L.NAMEPLATES_CONFIG_SHOW_PLAYER_NAMES_TITLE,
+				help = L.NAMEPLATES_CONFIG_SHOW_PLAYER_NAMES_HELP,
+				configKey = CONFIG_NAMEPLATES_SHOW_PLAYER_NAMES,
+				dependentOnOptions = {
+					CONFIG_NAMEPLATES_ENABLE_CUSTOMIZATIONS,
+				},
+			},
+			{
+				inherit = "TRP3_ConfigCheck",
+				title = L.NAMEPLATES_CONFIG_SHOW_PET_NAMES_TITLE,
+				help = L.NAMEPLATES_CONFIG_SHOW_PET_NAMES_HELP,
+				configKey = CONFIG_NAMEPLATES_SHOW_PET_NAMES,
+				dependentOnOptions = {
+					CONFIG_NAMEPLATES_ENABLE_CUSTOMIZATIONS,
+				},
+			},
+			{
+				inherit = "TRP3_ConfigCheck",
+				title = L.NAMEPLATES_CONFIG_SHOW_COLORS_TITLE,
+				help = L.NAMEPLATES_CONFIG_SHOW_COLORS_HELP,
+				configKey = CONFIG_NAMEPLATES_SHOW_COLORS,
+				dependentOnOptions = {
+					CONFIG_NAMEPLATES_ENABLE_CUSTOMIZATIONS,
+				},
+			},
+			{
+				inherit = "TRP3_ConfigCheck",
+				title = L.NAMEPLATES_CONFIG_SHOW_ICONS_TITLE,
+				help = L.NAMEPLATES_CONFIG_SHOW_ICONS_HELP,
+				configKey = CONFIG_NAMEPLATES_SHOW_ICONS,
+				dependentOnOptions = {
+					CONFIG_NAMEPLATES_ENABLE_CUSTOMIZATIONS,
+				},
+			},
+			{
+				inherit = "TRP3_ConfigCheck",
+				title = L.NAMEPLATES_CONFIG_SHOW_TITLES_TITLE,
+				help = L.NAMEPLATES_CONFIG_SHOW_TITLES_HELP,
+				configKey = CONFIG_NAMEPLATES_SHOW_TITLES,
+				dependentOnOptions = {
+					CONFIG_NAMEPLATES_ENABLE_CUSTOMIZATIONS,
+				},
+			},
+			{
+				inherit = "TRP3_ConfigCheck",
+				title = L.NAMEPLATES_CONFIG_SHOW_OOC_INDICATORS_TITLE,
+				help = L.NAMEPLATES_CONFIG_SHOW_OOC_INDICATORS_HELP,
+				configKey = CONFIG_NAMEPLATES_SHOW_OOC_INDICATORS,
+				dependentOnOptions = {
+					CONFIG_NAMEPLATES_ENABLE_CUSTOMIZATIONS,
+					CONFIG_NAMEPLATES_SHOW_PLAYER_NAMES,
+				},
+			},
+			{
+				inherit = "TRP3_ConfigDropDown",
+				title = L.NAMEPLATES_CONFIG_OOC_INDICATOR_TITLE,
+				help = L.NAMEPLATES_CONFIG_OOC_INDICATOR_HELP,
+				configKey = CONFIG_NAMEPLATES_OOC_INDICATOR,
+				dependentOnOptions = {
+					CONFIG_NAMEPLATES_ENABLE_CUSTOMIZATIONS,
+					CONFIG_NAMEPLATES_SHOW_PLAYER_NAMES,
+					CONFIG_NAMEPLATES_SHOW_OOC_INDICATORS,
+				},
+				listContent = {
+					{
+						L.CO_TOOLTIP_PREFERRED_OOC_INDICATOR_TEXT .. OOC_TEXT_INDICATOR,
+						"TEXT",
+					},
+					{
+						L.CO_TOOLTIP_PREFERRED_OOC_INDICATOR_ICON .. tostring(OOC_ICON_INDICATOR),
+						"ICON",
+					},
+				},
+				listWidth = nil,
+				listCancel = true,
+			},
+		}
+	});
+end
+
+-- Module registration.
+TRP3_API.module.registerModule({
+	name        = L.NAMEPLATES_MODULE_NAME,
+	description = L.NAMEPLATES_MODULE_DESCRIPTION,
+	version     = 1,
+	id          = "TRP3_NamePlates",
+	onInit      = function(...) return TRP3_NamePlates:OnInitialize(...); end,
+	onStart     = function(...) return TRP3_NamePlates:OnEnable(...); end,
```

Comment:
I don't like this. This not how all other modules are declared (we have local functions `onInit`) and this looks like just trying to retrofit a AceAddon-ish based system with `AddOn:OnEnable()` method with a self bound to the scope for no actual usage (your calls to self could use `TRP3_NamePlates` directly).

If you take the entire add-on and existing modules into consideration I think this is an anti pattern

Reply:
The module itself is stateful, hence the preference for `self` since I can easily get at it from a global context and debug the whole module via `/dump` and `/tinspect`.

There was definitely a borrowing of AceAddon terminology, but that's mostly because my initial design would have had the global "enable customizations" toggle also call the appropriate "OnDisable" function to fully turn off events/hooks.